app/assets/javascripts/components/shared/modal.js

```diff
@@ -0,0 +1,64 @@
+(function (App) {
+
+  var templates = {
+    default: '',
+    team: 'templates/shared/modal-team'
+  };
+
+  App.Component.Modal = Backbone.View.extend({
+
+    className: 'modal-container',
+
+    template: JST['templates/shared/modal'],
+
+    events: {
+      'click .veil': 'onCloseModal'
+    },
+
+    initialize: function (options) {
+      console.log(options);
+      this.options = options;
+
+      // Binded functions
+      this.onKeyDownBinded = this._onKeyDown.bind(this);
+
+      this._setVars();
+      this._setEventListeners();
+    },
+
+    _setVars: function () {
+      // children inherit this var and use it to render the content
+      this.content = {};
+      this.body = document.querySelector('body');
+    },
+
+    _setEventListeners: function () {
+      document.addEventListener('keydown', this.onKeyDownBinded, false);
+    },
+
+    _removeEventListeners: function () {
+      document.removeEventListener('keydown', this.onKeyDownBinded, false);
+    },
+
+    _onKeyDown: function (e) {
+      // 27 is ESC key
+      if (e.keyCode !== 27) return;
+      this.onCloseModal();
```

Comment:
You probably want to also add `e.preventDefault()`.

An example where it's needed: open your browser in fullscreen on your Mac, open the modal, and push the escape key. You'll see the window exit the fullscreen mode.